### DISCUSSION TITLE
How to display the progress bar of background function operation

### DISCUSSION THREAD

**Discussion by xiaozhiagi** [zauberzeug/nicegui#1408] [Q&A]
### Question

Click the button on the front end, and the backend function will start running. However, sometimes the running time may be a bit long and the running progress is unknown. Therefore, I hope to be able to display this running progress on the page

**Comment by rodja** [comment]
Have you seen our [progress example](https://github.com/zauberzeug/nicegui/blob/main/examples/progress/main.py)? Also make sure [you do not block the UI](https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-my-long-running-function-blocking-ui-updates).

**Reply by xiaozhiagi** [reply]
Here is an example. I hope to click 'calculate' and the page can display the progress bar of the trigger function in real-time

```python
# coding: utf-8
from nicegui import app, ui


result = {}
result['data'] = {}


@ui.page('/')
def main_page():

    def start_computation():
        a = 3
        b = 10
        c = a * b
        result['data']['c'] = c
        ui.open('/')

    ui.button('compute', on_click=start_computation)

    ui.markdown(f"{result['data']}")

ui.run()

```

**Reply by rodja** [reply]
The computation in your example is instantly so I do not see a point in showing a progress bar. Why are you loading `/` after the result is done?

**Reply by xiaozhiagi** [reply]
For example, if a triggered function runs for a long time and there is a progress bar prompt at this time, it will obviously make the user prefer it and avoid mistakenly thinking that there is no response after clicking

**Reply by rodja** [reply]
Have you read the [FAQ](https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-my-long-running-function-blocking-ui-updates)? You need to use async/await for cpu intensive tasks.

**Comment by xiaozhiagi** [comment]
Or can the program be displayed on the page using ui. pinner (size='lg ') to remind the user that the function is currently loading. I'm not sure if it's possible to provide code examples of how to implement ui. piner (size='lg ') on the page
